packages/flutter/lib/src/rendering/grid.dart

```diff
@@ -73,65 +273,78 @@ class RenderGrid extends RenderBox with ContainerRenderObjectMixin<RenderBox, Gr
       child.parentData = new GridParentData();
   }
 
+  Size _getIntrinsicSize(BoxConstraints constraints) {
+    return constraints.constrain(_delegate.getGridSpecification(constraints, childCount).gridSize);
+  }
+
   double getMinIntrinsicWidth(BoxConstraints constraints) {
     assert(constraints.isNormalized);
-    // We can render at any width.
-    return constraints.constrainWidth(0.0);
+    return _getIntrinsicSize(constraints).width;
   }
 
   double getMaxIntrinsicWidth(BoxConstraints constraints) {
     assert(constraints.isNormalized);
-    double maxWidth = childCount * _maxChildExtent;
-    return constraints.constrainWidth(maxWidth);
+    return _getIntrinsicSize(constraints).width;
```

Comment:
Sorry, I copy and pasted the wrong question here. The question is what is the smallest width that doesn't increase the height. For FixedColumnCountGridDelegate it's presumably the same as the min intrinsic width. For a delegate that tries to size things around text, then it depends on the longest line of text.